base/special/gamma.jl

```diff
@@ -155,3 +155,23 @@ Natural logarithm of the absolute value of the [`beta`](@ref)
 function ``\\log(|\\operatorname{B}(x,y)|)``.
 """
 lbeta(x::Number, w::Number) = lgamma(x)+lgamma(w)-lgamma(x+w)
+
+"""
+    lbinomial(n, k)
+
+Natural logarithm of the [`binomial`](@ref) coefficient.
+"""
+function lbinomial(n::T, k::T) where {T<:Integer}
+    const S = float(T)
+    (k <  0) && return typemin(S)
+    if n < 0
+       n = -n + k - 1
+    end
+    k > n && return typemin(S)
+    (k == 0 || k == n) && return zero(S)
+    (k == 1) && return log(abs(n))
```

Comment:
In the code for `binomial`, it looks like this is `log((isodd(k) ? -1 : 1) * n)`

Reply:
Actually, I guess this is equivalent, it just needs `isodd(k) && throw(DomainError())` in the `n < 0` branch?

Reply:
Well, it's typical for `lbinomial` to actually mean `log(abs(binomial))`. I didn't note that in the docstr however.